CDCgov/prime-reportstream
# Okta ClinetID fixes

This PR extracts the hard-coded value of `clientId` and pulls the proper ID based on its environment. The 11ty site has instruction to pull the from the env but default to the ReportStream production id. @sean-usds If this env var exists in prod we should make sure we coordinate with devops to change it to the proper id.

Test Steps:
1. pull `kevinhaube/2658-okta-clientid-env`
2. `yarn` and `yarn start`
3. New tab -> navigate to Okta admin panel -> People -> you
4. Remove yourself from the `DHPrimeAdmin` and any state groups (keep a list of these to add yourself back to)
5. Add yourself to the `_TEST` group that only has the ReportStream application (not test) assigned
6. Back to RS tab -> Login
7. You _should_ see an error about not having access to that application because it is pulling the dev/staging key, a.k.a. `ReportStream (test)` client id.
8. Add yourself back to your old groups and ensure you have access to `ReportStream (test)`.
9. Clear all local Okta data
10. Re-login with success (hopefully!)

## Changes
- `oktaConfig.ts` now uses env vars for `clientId`
- Okta config for 11ty now uses env var for `clientId` with the production ID as a default to replace the old dev/staging id

## Checklist

### Testing

### Process

## Fixes
- #2658

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
